assets/javascripts/kitten/components/expandable/expand-board.js

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from 'react'
+import Radium from 'radium'
+import PropTypes from 'prop-types'
+import { Button } from 'kitten/components/buttons/button'
+import { ArrowIcon } from 'kitten/components/icons/arrow-icon'
+import COLORS from 'kitten/constants/colors-config'
+
+const ExpandBoardButton = ({
+  children,
+  expandChildren,
+  expanded,
+  disabled,
+  style,
+  onClick,
+  ariaId,
+}) => {
+  const defaultExpandChildren = expandChildren ? expandChildren : children
+  const buttonStyles = expanded
+    ? { ...style.base, ...style.expanded }
+    : style.base
+
+  return (
+    <Button
+      icon
+      iconOnRight
+      size="big"
+      modifier="helium"
+      disabled={disabled}
+      aria-expanded={expanded}
+      aria-controls={ariaId}
+      style={buttonStyles}
+      onClick={onClick}
+    >
+      {expanded ? defaultExpandChildren : children}
+      <ArrowIcon
+        direction={expanded ? 'top' : 'bottom'}
+        className="k-Button__icon"
+        fill={COLORS.background1}
+        style={style.arrow}
+      />
+    </Button>
+  )
+}
+
+ExpandBoardButton.propTypes = {
+  children: PropTypes.node.isRequired,
+  expandChildren: PropTypes.node,
+  expanded: PropTypes.bool,
+  disabled: PropTypes.bool,
+  style: PropTypes.object,
+  onClick: PropTypes.func,
+  ariaId: PropTypes.string.isRequired,
+}
+
+ExpandBoardButton.defaultProps = {
+  expandChildren: null,
+  expanded: false,
+  disabled: false,
+  style: {},
+  onClick: () => {},
+  ariaId: 'k-ExpandBoard',
+}
+
+const ExpandBoardContent = ({ children, ariaId }) => (
+  <div id={ariaId}>{children}</div>
+)
+
+ExpandBoardContent.propTypes = {
+  children: PropTypes.node.isRequired,
+  ariaId: PropTypes.string,
+}
+
+ExpandBoardContent.defaultProps = {
+  ariaId: 'k-ExpandBoard',
+}
+
+class ExpandBoardBase extends Component {
+  static Button = ExpandBoardButton
+  static Content = ExpandBoardContent
+
+  constructor(props) {
+    super()
+    this.state = {
+      expanded: false,
+    }
+  }
+
+  isButtonComponent = component => component.type.name === 'ExpandBoardButton'
+  isContentComponent = component => component.type.name === 'ExpandBoardContent'
+
+  handleClick = () => {
+    const newState = { expanded: !this.state.expanded }
```

Comment:
Vu qu'on dépends de l'état, ce n'est pas mieux de passer à `setState` une fonction en premier argument qui prends l'état précédent ?

```suggestion
    const newState = prevState => { { expanded: !prevState.expanded } }
```

Reply:
Tout à fait ! Néanmoins, je ne peux pas appliquer ta suggestion car `newState` sert à la prop `onClick` également.

Reply:
Ah yes, zut. On est obligé du coup de dépendre du state au moment du `handleClick`. Ou alors la bonne pratique serait de déclencher le `onClick` après le `setState`, dans son callback avec le blur ?

Reply:
Oui je vais faire ça. Tu as raison, c'est plus robuste.